src/Generator/Generators/CSharp/CSharpMarshal.cs

```diff
@@ -358,6 +358,13 @@ private string HandleReturnedPointer(Class @class, string qualifiedClass)
                 Context.Before.WriteLine("else {0} = {1}.{2}({3});", ret, qualifiedClass,
                     Helpers.CreateInstanceIdentifier, Context.ReturnVarName);
             }
+            Class returnedClass;
+            var pointee = Context.ReturnType.Type.Desugar().GetFinalPointee().Desugar();
```

Comment:
This cannot work either because of typedef-ed function pointers. For those we use the name of the typedef for the name of the delegate and such built-in desugaring would always eliminate that name.